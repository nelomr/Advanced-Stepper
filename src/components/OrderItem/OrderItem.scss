.order-item {
    padding: 0 10px 15px;
    margin-bottom: 20px;
    border-bottom: 1px solid $grey-medium;

    @include breakpoint($min-desktop) {
        display: flex;
        justify-content: space-between;
    }
}

.order-item--content,
.order-item--button {
    @include breakpoint($min-desktop) {
        flex: 0 1 50%;
        width: 50%;
    }
}

.order-item--content {
    display: flex;
    margin-bottom: 10px;

    @include breakpoint($min-desktop) {
        margin-bottom: 0;
    }
}

.order-item--icon {
    display: flex;

    img {
        width: auto;
        height: 44px;
        margin-top: 0;
        border: none
    }
}

.order-item--text {
    margin-left: 20px;
    text-align: left;
}

.order-item--title {
    font-weight: bold;
}

.order-item--date {
    color: $text-color;
}

.order-item--button {
    text-align: right;

    .button {
        width: 100%;
        padding: 12px 30px;
        font-size: 14px;
        color: $light-color;
        cursor: pointer;
        cursor: hand;
        text-decoration: none;
        background-color: $secondary-color;
        appearance: none;
        border: none;
        border-radius: 4px;

        @include breakpoint($min-tablet) {
            width: auto;
        }
    }
}