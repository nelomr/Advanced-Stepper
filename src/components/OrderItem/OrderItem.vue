<template v-if="order">
  <div class="order">
    <div class="order--content">
      <div class="order--icon">
        <img v-if="order.logisticCode === 'DELIVERED-PICKUP'" src="@/assets/images/delivery.png" alt="marcador">
        <img v-else src="@/assets/images/marcador.png" alt="marcador">
      </div>
      <div class="order--text">
        <div class="order--title">Pedido nยบ {{ order.code }}</div>
        <div class="order--date">Fecha pedido: {{ order.orderDate }}</div>
      </div>
    </div>
      <div :id="order.code" class="order--button">
        <router-link class="button"
          :to="{name:'orderId', params: {id: order.code}}"
        >
          VER PEDIDO
        </router-link>
      </div>
  </div>
</template>

<script>

export default {
  name: 'order',
  props: {
    order: {
      type: Object,
      required: true
    }
  }
}
</script>

<style scoped lang="scss">
  @import "./OrderItem.scss";
</style>