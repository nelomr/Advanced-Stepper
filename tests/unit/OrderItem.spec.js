import {shallowMount, createLocalVue} from '@vue/test-utils';
import OrderItem from '@/components/OrderItem/OrderItem';
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

describe('OrderItem', () => {
    let props;
    let order = {"code":"CO107341","orderDate":"2018-11-09","id":"e29f59ab-2793-4644-bec5-5818fd67a165","logisticCode":"LOCAL","cuponId":"","totalPrice":14500,"totalDeliveryPrice":0,"productsOrder":[{"productId":"b6d1f4e9-85a9-449d-a16e-cb308cbbeb35","quantity":1},{"productId":"8bf0e74b-284f-4a51-bfb1-aa5f1e0ab279","quantity":1},{"productId":"46297509-f50e-4b19-ba6c-bf9744414f37","quantity":2}],"pickupDates":[{"id":"01e6ebbf-99fc-443d-9cb4-a1d0714f2892","pickupDate":"2018-11-13","pickupStart":"17:00:00","pickupEnd":"18:00:00"},{"id":"fe4d70df-7a7c-43dc-b45f-21c7e9ae55e1","pickupDate":"2018-11-09","pickupStart":"17:00:00","pickupEnd":"18:00:00"}]};
    
    beforeEach(() => {
        props = {
            order
        };
    });
    
    const build = () => {
        const wrapper = shallowMount(OrderItem, {
            propsData: props,
            router,
            stubs: ['router-link', 'router-view']
        });

        return {
            wrapper
        };
    };

    it('renders the component', () => {
        const {wrapper} = build();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
