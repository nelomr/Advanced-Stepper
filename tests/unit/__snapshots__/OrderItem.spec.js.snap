// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem renders the component 1`] = `
<div class="order">
  <div class="order--content">
    <div class="order--icon"><img src="@/assets/images/marcador.png" alt="marcador"></div>
    <div class="order--text">
      <div class="order--title">Pedido nยบ CO107341</div>
      <div class="order--date">Fecha pedido: 2018-11-09</div>
    </div>
  </div>
  <div class="order--button">
    <router-link-stub to="[object Object]" class="button">
      VER PEDIDO
    </router-link-stub>
  </div>
</div>
`;
